### PR TITLE
add to/of_cstruct to use in TLS

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

File: p256/point.ml [hunk 1]
```diff
@@ -66,6 +68,17 @@ let pp fmt p =
       Format.fprintf fmt "04%a%a" Cstruct_util.pp_hex_le x
         Cstruct_util.pp_hex_le y
 
+let to_cstruct p =
```
Comment (emillon) [inline p256/point.ml]: Can you update `pp` to use `to_cstruct`?

Comment (hannesm) [reply p256/point.ml]: done

File: p256/point.ml [hunk 2]
```diff
@@ -58,13 +60,18 @@ let to_affine p =
...
-      Format.fprintf fmt "04%a%a" Cstruct_util.pp_hex_le x
-        Cstruct_util.pp_hex_le y
+    let four = Cstruct.create 1 in
+    Cstruct.set_uint8 four 0 4 ;
+    let rev_x = Cstruct_util.rev x
+    and rev_y = Cstruct_util.rev y
+    in
+    Cstruct.concat [ four ; rev_x ; rev_y ]
+
+let pp fmt p = Cstruct.hexdump_pp fmt (to_cstruct p)
```
Comment (emillon) [inline p256/point.ml]: I'd like to keep the format for now, so something like that should work (I'll upstream `Cstruct_util` and clean this mess later):
```suggestion
let pp fmt p = Cstruct_util.pp_hex_le fmt (Cstruct_util.rev p)
```

**Comment by emillon** [comment]
Thanks!

**Pull request merged**